File: model/src/main/kotlin/net/dungeonhub/model/ScoreResetModel.java
package net.dungeonhub.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import me.taubsie.dungeonhub.common.DungeonHubService;
import me.taubsie.dungeonhub.common.entity.model.Model;

@Getter
@AllArgsConstructor
public class ScoreResetModel implements Model {
    private long defaultCount;
    private long eventCount;

    public static ScoreResetModel fromJson(String json) {
        return DungeonHubService.getInstance()
                .getGson()
                .fromJson(json, ScoreResetModel.class);
    }
}

File: model/src/main/kotlin/net/dungeonhub/model/score/ScoreResetModel.kt
package net.dungeonhub.model.score

import net.dungeonhub.service.MoshiService
import net.dungeonhub.structure.model.Model

class ScoreResetModel(
    val defaultCount: Long,
    val eventCount: Long
) : Model {
    companion object {
        fun fromJson(json: String): ScoreResetModel {
            return MoshiService.moshi.adapter(ScoreResetModel::class.java).fromJson(json)!!
        }
    }
}